website/docs/cdktf/debugging.mdx

```diff
@@ -0,0 +1,59 @@
+---
+page_title: Debugging
+description: >-
+  Learn how to troubleshoot problems in your CDKTF application.
+---
+
+# Debugging
+
+## Increasing the verbosity
+
+To get more information about what CDKTF is doing and what problems have occured by setting the `CDKTF_LOG_LEVEL` environment variable to `DEBUG`. If you want to share the logs you can set `CDKTF_LOG_FILE_DIRECTORY` to a file path, a `cdktf.log` file with all logs will be written.
+
+## Troubleshooting
+
+To understand where the problem is coming from it's a good first step to see if it's a synth-time issue or an execution-time issue.
+If `cdktf synth` fails you have a synth-time issue, otherwise it's an execution-time issue.
+
+### Synth-time issues
+
+The most important thing is to understand which construct is causing the problem. The error message thrown might hint you towards it, if not you will need to search for it. If you have a last version that successfully synthesized you can search through the new changes, otherwise you need to search through all changes. To search it you need to repeatedly run `cdktf synth` and comment parts out until you find the error. The synthesized code does not need to make any logical sense since we don't plan to apply it.
```

Comment:
```suggestion
The most important thing is to understand which construct is causing the problem. The error message thrown might hint you towards it, if not you will need to search for it. If you have a last version that successfully synthesized you can search through the new changes, otherwise you need to search through all changes. To search it you can repeatedly run `cdktf synth` and comment parts out until you find the error. The synthesized code does not need to make any logical sense since we don't plan to apply it.
```

nit: need to sounds a bit harsh, but might just be me